### PR TITLE
Rationalize API exports in torch_python

### PR DESCRIPTION
**Pull request by peterbell10**
This renames `WindowsTorchApiMacro.h` to `Export.h` to mirror the c10 header `c10/macros/Export.h` and also updates it to use `C10_EXPORT`/`C10_IMPORT`. This also removes the `THP_API` macro from `THP_export.h` which appears to serve the same purpose.

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

### CODE REVIEW

**Comment by peterbell10** [comment]
@albanD would you mind taking a look?

File: torch/csrc/Generator.h [hunk 1]
```diff
@@ -14,16 +14,16 @@ struct THPGenerator {
...
 // borrowed. The caller should ensure that the at::Generator object lifetime
 // last at least as long as the Python wrapper.
-THP_API PyObject * THPGenerator_initDefaultGenerator(at::Generator cdata);
+TORCH_PYTHON_API PyObject * THPGenerator_initDefaultGenerator(at::Generator cdata);
 
 #define THPGenerator_Check(obj) \
   PyObject_IsInstance(obj, THPGeneratorClass)
 
-THP_API PyObject *THPGeneratorClass;
+TORCH_PYTHON_API extern PyObject *THPGeneratorClass;
```
Comment (albanD) [inline torch/csrc/Generator.h:22]: Why do you need to re-add the extern here? (and in other places)

File: torch/csrc/WindowsTorchApiMacro.h [hunk 2]
```diff
@@ -1,9 +1,2 @@
 #pragma once
-
-#include <c10/macros/Export.h>
-
-#ifdef _WIN32
-#define TORCH_PYTHON_API
-#else
-#define TORCH_PYTHON_API TORCH_API
-#endif
+#include <torch/csrc/Export.h>
```
Comment (albanD) [inline torch/csrc/WindowsTorchApiMacro.h:2]: Should we just delete this file? (can be in a follow up if you prefer)

Comment (peterbell10) [reply torch/csrc/WindowsTorchApiMacro.h:2]: We should, but it's included everywhere in the torch folder so the diff is massive. Can do it in separate PR though.

File: torch/csrc/Generator.h (hunk 1 above)
Comment (peterbell10) [reply torch/csrc/Generator.h:22]: `extern` is necessary for variables defined in another file, but it doesn't apply to classes or structs which is why  there were two macros: `THP_API` and `THP_CLASS`. Now we just have one and manually add `extern`. This is consistent with how all the other `_API` macros work.

File: torch/csrc/WindowsTorchApiMacro.h (hunk 2 above)
Comment (albanD) [reply torch/csrc/WindowsTorchApiMacro.h:2]: Follow up PR sounds good!

**Comment by peterbell10** [comment]
Rebased to fix merge conflict.

**Review by albanD — APPROVED** [review]
Thanks

**Pull request closed without merging**